pkg/sources/chunker.go

```diff
@@ -37,10 +37,10 @@ func Chunker(originalChunk *Chunk) chan *Chunk {
 			}
 			peekData, _ := reader.Peek(PeekSize)
 			chunk.Data = append(chunkBytes[:n], peekData...)
-			chunkChan <- &chunk
 			if errors.Is(err, io.EOF) {
 				break
 			}
+			chunkChan <- &chunk
```

Comment:
**Question:** I guess this is implementation dependent, but could we have read data _and_ gotten an EOF error? Maybe we should emit the chunk only if `n > 0`.

Reply:
Good call. Better to have the catch there just in case.